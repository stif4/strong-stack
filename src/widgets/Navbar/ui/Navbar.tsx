import { classNames } from 'shared/helpers/classNames/classNames';
import cls from './Navbar.module.scss';
import { Button, ButtonTheme } from 'shared/ui/Button/Button';
import { useTranslation } from 'react-i18next';
import { useCallback, useState } from 'react';
import { Modal } from 'shared/ui/Modal/Modal';

interface NavbarProps {
    className?: string;
}

export const Navbar = ({ className }: NavbarProps) => {
    const { t } = useTranslation();
    const [isAuthModal, setIsAuthModal] = useState(false);

    const onToggleModal = useCallback(() => {
        setIsAuthModal((prev) => !prev);
    }, []);

    return (
        <div className={classNames(cls.navbar, {}, [className])}>
            <Button
                theme={ButtonTheme.CLEAR_INVERTED}
                className={cls.links}
                onClick={onToggleModal}
            >
                {t('Войти')}
            </Button>
            <Modal isOpen={isAuthModal} onClose={onToggleModal}>
                {}
                {t(
                    'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid commodi consequatur eligendi impedit incidunt necessitatibus possimus quis saepe sunt totam.',
                )}
            </Modal>
        </div>
    );
};
