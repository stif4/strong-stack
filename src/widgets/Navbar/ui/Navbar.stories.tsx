import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import { Navbar } from './Navbar';

export default {
    title: 'widget/Navbar',
    component: Navbar,
} satisfies Meta<typeof Navbar>;

const Template: StoryFn<typeof Navbar> = (args) => <Navbar {...args} />;

export const Light = Template.bind({});
Light.args = {};
