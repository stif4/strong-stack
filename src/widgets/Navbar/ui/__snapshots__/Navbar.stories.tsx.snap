// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`widget/Navbar Light smoke-test 1`] = `
<div class="app dark">
  <div class="src-widgets-Navbar-ui-Navbar-module__navbar">
    <div class="src-widgets-Navbar-ui-Navbar-module__links">
      <a class="src-shared-ui-AppLink-AppLink-module__appLink src-widgets-Navbar-ui-Navbar-module__mainLink src-shared-ui-AppLink-AppLink-module__secondary"
         href="/"
         data-discover="true"
      >
        Главная
      </a>
      <a class="src-shared-ui-AppLink-AppLink-module__appLink src-shared-ui-AppLink-AppLink-module__red"
         href="/about"
         data-discover="true"
      >
        О сайте
      </a>
    </div>
  </div>
</div>
`;
